core/app/models/spree/image.rb

```diff
@@ -14,7 +14,7 @@ class Image < Asset
 
     # save the w,h of the original image (from which others can be calculated)
     # we need to look at the write-queue for images which have not been saved yet
-    after_post_process :find_dimensions
+    before_save :find_dimensions
```

Comment:
@aishmita-aggarwal @priyank-gupta - I guess we should not run this callback every time an Image object is updated, we should run this only when an attachment is updated in Image object.